app/uk/gov/hmrc/nationaldutyrepaymentcenter/controllers/ClaimController.scala

```diff
@@ -66,54 +67,82 @@ class ClaimController @Inject()(
         claimService.createClaim(eisCreateCaseRequest, correlationId).flatMap {
           case success: EISCreateCaseSuccess =>
             transferFilesToPega(success.CaseID, correlationId, createCaseRequest.uploadedFiles)
-              .map { fileTransferResults =>
-                val response = NDRCCreateCaseResponse(
-                    correlationId = correlationId,
-                    result = Option(
-                      NDRCFileTransferResult(success.CaseID, LocalDateTime.now(), fileTransferResults)
-                    ))
-               Created(Json.toJson(response))
+              .flatMap { fileTransferResults =>
+                val response = NDRCCaseResponse(
+                  correlationId = correlationId,
+                  result = Option(
+                    NDRCFileTransferResult(success.CaseID, LocalDateTime.now(), fileTransferResults)
+                  ))
+                auditService
+                  .auditCreateCaseEvent(createCaseRequest)(response)
+                  .map(_ => Created(Json.toJson(response)))
               }
 
           // when request to the upstream api returns an error
           case error: EISCreateCaseError =>
-              Future.successful(BadRequest(
-                Json.toJson(
-                  NDRCCreateCaseResponse(
-                    correlationId = correlationId,
-                    error = Some(
-                      ApiError(
-                        errorCode = error.errorCode.getOrElse("ERROR_UPSTREAM_UNDEFINED"),
-                        errorMessage = error.errorMessage
-                      )
-                    )
+            if (error.isDuplicateCaseError) {
```

Comment:
@chandaksweta we don't have duplicate case errors on NDRC. Can we remove this?

Reply:
For createCase what happens if the user goes back in the browser and submits the same request again??

Reply:
and again...